packages/auth/src/client/service.ts

```diff
@@ -88,7 +150,44 @@ async function verifySig(
 ): Promise<boolean> {
   const publicKey = (await importJWK(jwk)) as crypto.KeyLike
   const data = Buffer.from(challenge)
-  return crypto.verify(null, data, publicKey, Buffer.from(sig))
+  return crypto.verify(null, data, publicKey, Buffer.from(sig, 'base64'))
+}
+
+async function verifySigFromBoundKey(
+  deps: ServiceDependencies,
+  sig: string,
+  sigInput: string,
+  grant: Grant,
+  ctx: AppContext
+): Promise<VerifySigFromBoundKeyResult> {
+  const registryData = await getRegistryDataByKid(deps, grant.clientKeyId)
+  if (!registryData)
+    return {
+      success: false,
+      error: 'invalid_client',
+      status: 401
+    }
+  const { keys } = registryData
+  const clientKey = keys[0]
```

Comment:
Why is that an array? Should there be just one key per key id?

Reply:
Looking back, I think this was the result of a misinterpretation of Adrian's original doc. Which is probably pretty moot now that the keys aren't being provided by an external registry anymore.

Reply:
After discussing with Adrian on a new `kid` response, we settled on the shape described [here](https://github.com/interledger/rafiki/issues/424). I'll move that work to outside this PR so that it doesn't grow any larger than it already has, but the end result will be that we won't need to handle an array anymore.